recipe/cross_compile_support.sh

```diff
@@ -15,19 +19,52 @@ if [[ "${HOST_PLATFORM}" != "${BUILD_PLATFORM}" ]]; then
             for pkg in $(cat ${CI_SUPPORT}/../recipe/yum_requirements.txt); do
                 if [[ "${pkg}" != "#"* && "${pkg}" != "" ]]; then
-                    mamba install "${pkg}-cos7-${HOST_PLATFORM:6}" -n sysroot_${HOST_PLATFORM} --yes --quiet || true
+                    mamba install "${pkg}-cos7-${HOST_PLATFORM_ARCH}" -n sysroot_${HOST_PLATFORM} --yes --quiet || true
                 fi
             done
         fi
         export QEMU_LD_PREFIX=$(find ${CONDA_PREFIX}/envs/sysroot_${HOST_PLATFORM} -name sysroot | head -1)
         if [ -f ${CI_SUPPORT}/${CONFIG}.yaml ]; then
             echo "CMAKE_CROSSCOMPILING_EMULATOR: " >> ${CI_SUPPORT}/${CONFIG}.yaml
-            echo "- /usr/bin/qemu-$(echo $HOST_PLATFORM | cut -b 7-)-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+            echo "- /usr/bin/qemu-${HOST_PLATFORM_ARCH}-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
             echo "CROSSCOMPILING_EMULATOR: " >> ${CI_SUPPORT}/${CONFIG}.yaml
-            echo "- /usr/bin/qemu-$(echo $HOST_PLATFORM | cut -b 7-)-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+            echo "- /usr/bin/qemu-${HOST_PLATFORM_ARCH}-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+        fi
+
+
+        if [[ "${CUDA_COMPILER_VERSION}" == "11.2" ]] && [[ "${CDT_NAME}" == "cos8" || "${CDT_NAME}" == "cos7" ]]; then
+            # We use cdt_name=cos7 for rhel8 based nvcc till we figure out
+            # a stable cos8 replacement.
+            EXTRACT_DIR=$(mktemp -d)
+            pushd ${EXTRACT_DIR}
+                if [[ "${HOST_PLATFORM_ARCH}" == "aarch64" ]]; then
+                    CUDA_HOST_PLATFORM_ARCH=sbsa
+                else
+                    CUDA_HOST_PLATFORM_ARCH=${HOST_PLATFORM_ARCH}
+                fi
+                curl -L https://developer.download.nvidia.com/compute/cuda/repos/rhel8/sbsa/version_11.2.2.json > versions.json
+                # read names & versions for necessary RPMs from manifest; download & install them
+                jq 'keys[] as $k | "\($k)-11-2-\(.[$k] | .version)-1"' versions.json | while read d; do
+                    # normalize "_" -> "-"; also adapt "_dev" -> "-devel" (for cuda_nvml_dev)
+                    fn="$(echo $d | sed 's/_/-/g' | sed 's/-dev/-devel/g').${HOST_PLATFORM_ARCH}.rpm"
+                    curl -L -O https://developer.download.nvidia.com/compute/cuda/repos/rhel8/${CUDA_HOST_PLATFORM_ARCH}/${fn}
+                    bsdtar -xvf ${fn}
+                done
+                mv ./usr/local/cuda-${CUDA_COMPILER_VERSION}/targets/${CUDA_HOST_PLATFORM_ARCH}-linux ${CUDA_HOME}/targets/${CUDA_HOST_PLATFORM_ARCH}-linux
+            popd
+            rm -rf ${EXTRACT_DIR}
+        elif [[ "${CUDA_COMPILER_VERSION}" == "11.2" ]]; then
+	    echo "cross compiling with cuda == 11.2 and cdt != cos7/8 not supported yet"
+	    exit 1
+        elif [[ "${CUDA_COMPILER_VERSION}" != "None" ]]; then
+	    # FIXME: can use anaconda.org/nvidia packages to get the includes and libs
+	    # for cuda >=11.3.
```

Comment:
Do we care about supporting cross compilation for CUDA 11.0 or 11.1? My take is probably not, but figure I'd ask.

Reply:
I'd say only 11.2+